Source/ACE.Entity/AceObject.cs

```diff
@@ -1816,24 +1816,48 @@ public object Clone(uint guid)
         {
             AceObject ret = (AceObject)Clone();
             ret.AceObjectId = guid;
+            // We are cloning a new AceObject with a new AceObjectID - need to set this to false. Og II
+            ret.HasEverBeenSavedToDatabase = false;
+            ret.IsDirty = true;
+
+            ret.PaletteOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.TextureOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.AnimationOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.IntProperties.ForEach(c => c.AceObjectId = guid);
+            ret.Int64Properties.ForEach(c => c.AceObjectId = guid);
+            ret.DoubleProperties.ForEach(c => c.AceObjectId = guid);
+            ret.BoolProperties.ForEach(c => c.AceObjectId = guid);
+            ret.DataIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.InstanceIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.StringProperties.ForEach(c => c.AceObjectId = guid);    
+            ret.GeneratorLinks.ForEach(c => c.AceObjectId = guid);
+            
+            // We're not seemingly changing the guids for the Dictionaries below from Clone(), is this bad?
```

Comment:
@ddevec you and @Mogwai-TheFurry  need to talk.   When we clone a new object from a weenie, we have to do a deep reset of those values.   I don't care where it takes place, but it has to happen prior to save.   I have this in my code.   I made it part of the clone new method.   It happens behind the scenes.